x-pack/plugins/security_solution/public/timelines/components/timeline/helpers.tsx

```diff
@@ -253,3 +274,57 @@ export const focusUtilityBarAction = (containerElement: HTMLElement | null) => {
 export const resetKeyboardFocus = () => {
   document.querySelector<HTMLAnchorElement>('header.headerGlobalNav a.euiHeaderLogo')?.focus();
 };
+
+const buildISQueryMatch = ({
+  browserFields,
+  field,
+  isFieldTypeNested,
+  value,
+}: {
+  browserFields: BrowserFields;
+  field: string;
+  isFieldTypeNested: boolean;
+  value: string | number;
+}): string => {
+  if (isFieldTypeNested) {
+    return convertNestedFieldToQuery(field, value, browserFields);
+  } else if (checkIfFieldTypeIsDate(field, browserFields)) {
+    return convertDateFieldToQuery(field, value);
+  } else {
+    return `${field} : ${isNumber(value) ? value : escapeQueryValue(value)}`;
+  }
+};
+
+const buildEXISTSQueryMatch = ({
+  browserFields,
+  field,
+  isFieldTypeNested,
+}: {
+  browserFields: BrowserFields;
+  field: string;
+  isFieldTypeNested: boolean;
+}): string => {
+  return isFieldTypeNested
+    ? convertNestedFieldToExistQuery(field, browserFields)
+    : `${field} ${EXISTS_OPERATOR}`;
+};
+
+const buildISONEOFQueryMatch = ({
```

Comment:
Consider adding a unit test for the `combineQueries` function (defined in `x-pack/plugins/timelines/public/components/t_grid/helpers.tsx`) that asserts both the data provider (with an `IS_ONE_OF` operator) shown in the screenshot below, and the (KQL) filter representation (also in the screenshot below), are both converted to the same ES query DSL:

![assert-same-query-dsl](https://user-images.githubusercontent.com/4459398/193946302-62f114a3-1d41-4298-b263-3ed206a4273b.png)